### PR TITLE
upgrade(installer): Uninstall deb/rpm agent when installing the OCI

### PR DESCRIPTION
**Pull request by BaptisteFoy**
### What does this PR do?
Uninstalls the non-managed agent when installing the managed agent

### Motivation
Avoid clogging disk space on customer infrastructure

### Describe how you validated your changes
E2E + manual testing on linux VMs

### Additional Notes

### CODE REVIEW

**Review by coignetp — APPROVED** [review]

File: pkg/fleet/installer/packages/pkg_manager.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+			cmd = exec.Command("rpm", "-e", "--nodeps", agentPackage)
+		}
+	}
+
+	if cmd == nil {
+		// If we can't find a package manager we'll simply remove the directory
+		// where the package is expected to be installed
+		switch pkg {
+		case agentPackage:
+			cmd = exec.Command("rm", "-rf", pathOldAgent)
```
Comment (arbll) [inline pkg/fleet/installer/packages/pkg_manager.go:42]: If we can't find the package manager let's simply skip this removal

File: test/new-e2e/tests/installer/unix/package_agent_test.go [hunk 2]
```diff
@@ -422,14 +422,15 @@ func (s *packageAgentSuite) TestUpgrade_DisabledAgentDebRPM_to_OCI() {
...
-	s.host.AssertPackageInstalledByPackageManager("datadog-agent")
+	s.host.AssertPackageNotInstalledByPackageManager("datadog-agent")
 
 	s.host.Run("sudo systemctl show datadog-agent -p ExecStart | grep /opt/datadog-packages")
 }
 
 func (s *packageAgentSuite) TestInstallWithLeftoverDebDir() {
 	// create /opt/datadog-agent to simulate a disabled agent
 	s.host.Run("sudo mkdir -p /opt/datadog-agent")
+	defer func() { s.host.Run("sudo rm -rf /opt/datadog-agent") }()
```
Comment (arbll) [inline test/new-e2e/tests/installer/unix/package_agent_test.go:433]: Why is this needed ?

File: pkg/fleet/installer/packages/pkg_manager.go [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+// removeDebRPMPackage removes a package installed via deb/rpm package manager
+// It doesn't remove dependencies or purge as we want to keep existing configuration files
+// and reinstall the package using the installer.
+func removeDebRPMPackage(ctx context.Context, pkg string) (err error) {
+	span, _ := tracer.StartSpanFromContext(ctx, "remove_deb_rpm_package")
+	defer func() { span.Finish(tracer.WithError(err)) }()
+	// Compute the right command depending on the package manager
+	var cmd *exec.Cmd
+	if _, pathErr := exec.LookPath("dpkg"); pathErr == nil {
+		cmd = exec.Command("dpkg", "-r", agentPackage)
```
Comment (arbll) [inline pkg/fleet/installer/packages/pkg_manager.go:28]: I assume we're not querying dpkg because it always succeeds on a remove even if the package is not installed ?

File: pkg/fleet/installer/packages/pkg_manager.go [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+	} else if _, pathErr := exec.LookPath("rpm"); pathErr == nil {
+		// Check if package exist, else the command will fail
+		pkgErr := exec.Command("rpm", "-q", agentPackage).Run()
+		if pkgErr == nil {
+			cmd = exec.Command("rpm", "-e", "--nodeps", agentPackage)
```
Comment (raphaelgavache) [inline pkg/fleet/installer/packages/pkg_manager.go:33]: I wonder if we can hit some bad surprises in the prerm/postrm scripts of the different agent versions you can uninstall, is it possible to skip those scripts with the uninstall?

Comment (raphaelgavache) [reply pkg/fleet/installer/packages/pkg_manager.go:33]: ex [noscripts](http://ftp.rpm.org/max-rpm/s1-rpm-install-additional-options.html#S2-RPM-INSTALL-NOSCRIPTS)

File: test/new-e2e/tests/installer/unix/package_agent_test.go (hunk 2 above)
Comment (BaptisteFoy) [reply test/new-e2e/tests/installer/unix/package_agent_test.go:433]: The test doesn't clean up properly if it fails and fails the next tests. I had issues debugging because of it.

File: pkg/fleet/installer/packages/pkg_manager.go (hunk 3 above)
Comment (BaptisteFoy) [reply pkg/fleet/installer/packages/pkg_manager.go:28]: Correct

File: pkg/fleet/installer/packages/pkg_manager.go (hunk 4 above)
Comment (BaptisteFoy) [reply pkg/fleet/installer/packages/pkg_manager.go]: That's a good point; I agree and I'll add it.

For completeness I checked the prerm/postrm; there are a few caveats in not running them:
- Some files not managed by apt will be leftover in `/opt/datadog-agent`; such as the RC DB, custom integrations...
- The install method will be kept and the rest of the install will not override it - wrongly reporting dpkg / rpm

**Review by arbll — APPROVED** [review]

File: pkg/fleet/installer/packages/pkg_manager.go (hunk 4 above)
Comment (raphaelgavache) [reply pkg/fleet/installer/packages/pkg_manager.go]: The install method you're referring is the debug telemetry file?

Comment (BaptisteFoy) [reply pkg/fleet/installer/packages/pkg_manager.go]: We use it in Fleet to check if an agent can be upgraded; so it's not just telemetry. Though it should be fine here as it's overwritten by the install script.
